Source/Core/VideoCommon/TextureCacheBase.cpp

```diff
@@ -542,32 +555,39 @@ class ArbitraryMipmapDetector
     u32 row_length;
     const u8* buffer;
 
-    PixelRGBAf Sample(u32 x, u32 y) const
+    u32 byte_size() const
+    {
+      return row_length * height * 4;
+    }
+
+    static PixelRGBAf Sample(const u8* src, const Level& src_shape, u32 x, u32 y)
```

Comment:
Just curious, why call it `src_shape`?  Is this the mip level texture?

Reply:
So the `src` is the downsample buffer, but the downsample buffer is "shaped" (has same width, height and row_length) than the mip. So I'm passing in the mip info directly to make sure they're the same. So yeah, you're not supposed to use the buffer in there I guess.

Reply:
Changed it to take a `Shape` object so it's cleaner.